### DISCUSSION TITLE
Group Toggle

### DISCUSSION THREAD

**Discussion by xHirscHx** [HASwitchPlate/HASPone#59] [General]
So I finally got everything to the way I had it setup on the old firmware except the group toggle. p3 b4  I cant use a scene because the lights are on a Sonoff 4ch pro and if you tried it you know what i mean. Any help would be great
THX.

**Comment by aderusha** [comment]
I'm not sure if I get what you're after here.  Can you make a group and use that as the entity you toggle?  Why won't scenes work?

**Reply by xHirscHx** [reply]
Im looking for something like with the old setup page 3 were button 7  would toggle all lights off if 1 was on . Or toggle all on. I have a group called (den toggle group). The sonoff 4ch pro is hard to set up scenes with because its one device with 4 entries. I've tried before and have had the lights go into a loop (Disco style). I guess i'll give it another go. What i need is to get this to work.

This is what's  running anther plate running old firmware

  # Toggle all light_toggle_group when p[3].b[7] pressed
  - alias: hasp_cjs_plate_p3_ToggleAll
    trigger:
      - platform: mqtt
        topic: "hasp/cjs_plate/state/p[3].b[7]"
        payload: "ON"
    action:
      - service: >
          {% if states('group.cjs_light_toggle_group') == "on" -%}
            homeassistant.turn_off
          {%- else -%}
            homeassistant.turn_on
          {%- endif %}
        entity_id: group.cjs_light_toggle_group

  # Set button 4 to "All Lights Off" if any lights are on, otherwise set to "All Lights On"
  - alias: hasp_cjs_plate_p3_ToggleColorAll
    trigger:
      - platform: state
        entity_id: group.cjs_light_toggle_group
      - platform: state
        entity_id: "binary_sensor.cjs_plate_connected"
        to: "on"
      - platform: homeassistant
        event: start
      - platform: state
        entity_id: input_number.hasp_cjs_plate_selectedbackgroundcolor
      - platform: state
        entity_id: input_number.hasp_cjs_plate_unselectedbackgroundcolor
      - platform: state
        entity_id: input_number.hasp_cjs_plate_selectedforegroundcolor
      - platform: state
        entity_id: input_number.hasp_cjs_plate_unselectedforegroundcolor
    action:
      - service: mqtt.publish
        data:
          topic: "hasp/cjs_plate/command/json"
          payload_template: >-
            [{% if states('group.cjs_light_toggle_group') == "on" -%}
              "p[3].b[7].txt=\"All Lights Off\"","p[3].b[7].bco={{ states('input_number.hasp_cjs_plate_unselectedbackgroundcolor') | int }}","p[3].b[7].pco={{ states('input_number.hasp_cjs_plate_unselectedforegroundcolor') | int }}"
            {%- else -%}
              "p[3].b[7].txt=\"All Lights On\"","p[3].b[7].bco={{ states('input_number.hasp_cjs_plate_selectedbackgroundcolor') | int }}","p[3].b[7].pco={{ states('input_number.hasp_cjs_plate_selectedforegroundcolor') | int }}"
            {%- endif %}]

**Reply by aderusha** [reply]
In that case, I'm pretty confident that a [group](https://www.home-assistant.io/integrations/group/) should give you what you want. Create the group, and have HASPone toggle that group.  You can adjust if a group is `On` if just one of the entities is on, or set it up such that all entities must be `on` for the group to be `on`.  Toggling a group will set all members to on or off.
